File: subprojects/shared/src/main/groovy/com/canoo/dolphin/core/PresentationModel.java
package com.canoo.dolphin.core;

import java.util.List;

public interface PresentationModel extends Observable {
    String DIRTY_PROPERTY = "dirty";

    String getId();

    List<Attribute> getAttributes();

    Attribute findAttributeByPropertyName(String propertyName);

    Attribute findAttributeByQualifier(String qualifier);

    Attribute findAttributeById(long id);

    void syncWith(PresentationModel other);

    String getPresentationModelType();

    void addAttribute(Attribute attribute);

	boolean isDirty();
}

File: dolphin-grails/web-app/src/comm/PresentationModel.js
define(function () {

    function PresentationModel(id, type) {

        this.id = id || PresentationModel.nextId();
        this.presentationModelType = type;

        this.attributes = [];

        this.addAttribute = function(attribute) {
            this.attributes.push(attribute);
        }

    }

    PresentationModel.nextId = (function() {
        var id = 0;
        return function() {
            return id++;
        }
    })();

    return PresentationModel;

});
